ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionCount.java

```diff
@@ -45,14 +62,48 @@ public String getName() {
 
   @Override
   public JsonRpcResponse response(final JsonRpcRequest request) {
-    if (request.getParamLength() != 2) {
-      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
-    }
-
     final Address address = parameters.required(request.getParams(), 0, Address.class);
-    final String privacyGroupId = parameters.required(request.getParams(), 1, String.class);
+    final String privacyGroupId;
+    long nonce = 0;
+    try {
+      if (request.getParamLength() == 2) {
+        privacyGroupId = parameters.required(request.getParams(), 1, String.class);
+      } else if (request.getParamLength() == 3) {
+        final String privateFrom = parameters.required(request.getParams(), 1, String.class);
+        final String[] privateFor = parameters.required(request.getParams(), 2, String[].class);
+        privacyGroupId = determineLegacyPrivacyGroupId(privateFrom, privateFor);
+      } else {
+        return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+      }
+      if (privacyGroupId != null) {
+        nonce = privateTransactionHandler.getSenderNonce(address, privacyGroupId);
+      }
+      return new JsonRpcSuccessResponse(request.getId(), Quantity.create(nonce));
```

Comment:
`nonce` will always be zero here when using the legacy privacy (`privateFrom` / `privateFor`), is that intentional? ...does the calculation of the nonce happen elsewhere?